torch/_inductor/codegen/cpp_gemm_template.py

```diff
@@ -286,22 +310,27 @@ def normalize_shapes(inputs, layout_or_out):
         new_inputs, _ = normalize_shapes(
             *maybe_to_dense(*reorder_and_filter(input_nodes, layout))
         )
-        m, n, k, *_ = mm_args(new_inputs[0], new_inputs[1])
+        int8_gemm = _is_int8_gemm(new_inputs)
+        m, n, k, *_ = mm_args(new_inputs[0], new_inputs[3 if int8_gemm else 1])
         micro_gemm = create_micro_gemm(
             "micro_gemm",
             m,
             n,
             k,
-            input_dtype=layout.dtype,
-            output_dtype=torch.float,
+            input_dtype=new_inputs[0].get_dtype() if int8_gemm else layout.dtype,
```

Comment:
seems `input_dtype=new_inputs[0].get_dtype()` always holds regardless of int8 gemm or not?

Reply:
I guess so, using `new_inputs[0].get_dtype()` as `input_dtype`.